src/context/UserTokenBalanceContext.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {
+  ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+} from 'react';
+import { GetUserTokenBalanceReturn, useGetUserTokenBalanceQuery } from '~gql';
+import { useAppContext, useColonyContext } from '~hooks';
+
+export const UserTokenBalanceContext = createContext<{
+  tokenBalanceData: GetUserTokenBalanceReturn | null | undefined;
+  pollActiveTokenBalance: () => void;
+  pollLockedTokenBalance: () => void;
+}>({
+  tokenBalanceData: null,
+  pollActiveTokenBalance: () => {},
+  pollLockedTokenBalance: () => {},
+});
+
+export const UserTokenBalanceProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const { wallet } = useAppContext();
+  const { colony } = useColonyContext();
+  const { colonyAddress, nativeToken } = colony || {};
+
+  const { data: tokenBalanceQueryData, refetch } = useGetUserTokenBalanceQuery({
+    variables: {
+      input: {
+        walletAddress: wallet?.address ?? '',
+        tokenAddress: nativeToken?.tokenAddress ?? '',
+        colonyAddress: colonyAddress ?? '',
+      },
+    },
+    skip: !wallet?.address || !nativeToken?.tokenAddress,
+  });
+
+  const tokenBalanceData = tokenBalanceQueryData?.getUserTokenBalance;
+
+  const pollActiveTokenBalance = useCallback(() => {
+    let interval;
+
+    const fetchActiveBalance = async () => {
+      const { data } = await refetch();
+      const activeBalance = data.getUserTokenBalance?.activeBalance;
+      if (activeBalance !== tokenBalanceData?.activeBalance) {
+        clearInterval(interval);
+      }
+    };
+
+    interval = setInterval(fetchActiveBalance, 3_000);
+  }, [tokenBalanceData, refetch]);
+
+  const pollLockedTokenBalance = useCallback(async () => {
+    let interval;
+
+    const fetchLockedBalance = async () => {
```

Comment:
Distinguishing between active and locked.

When staking / claiming, we can stop polling once the locked tokens value has changed.

When activating / deactivating tokens via the popover, the locked tokens value won't change, and therefore we won't know to stop polling if we're only looking for the locked value to change.